dependency_test.go

```diff
@@ -0,0 +1,4612 @@
+	 *                                         |
+	 *                           _____________j(19)___________
+	 *                          /                  |     |    \
+	 *         _______________p(31)____________   h(15) f(11) e(9)
+	 *        /         |      |     |     |   \
+	 *     _m(25)_     o(29) n(27) l(23) i(17) b(3)
+	 *    /       \
+	 *  a(1)     k(21)
+	 *   |
+	 *  d(7)
+	 *
+	 * Intermediate:
+	 *
+	 *                                f(11)   g(13)
+	 *                                         |
+	 *                                        c(5)
+	 *                                         |
+	 *                           _____________j(19)______
+	 *                          /                  |     \
+	 *         _______________p(31)____________   h(15) e(9)
+	 *        /         |      |     |     |   \
+	 *     _m(25)_     o(29) n(27) l(23) i(17) b(3)
+	 *    /       \
+	 *  a(1)     k(21)
+	 *   |
+	 *  d(7)
+	 *
+	 * Non-exclusive final:
+	 *
+	 *                                       f(11)
+	 *                                         |
+	 *                                       g(13)
+	 *                                         |
+	 *                                        c(5)
+	 *                                         |
+	 *                           _____________j(19)______
+	 *                          /                  |     \
+	 *         _______________p(31)____________   h(15) e(9)
+	 *        /         |      |     |     |   \
+	 *     _m(25)_     o(29) n(27) l(23) i(17) b(3)
+	 *    /       \
+	 *  a(1)     k(21)
+	 *   |
+	 *  d(7)
+	 */
+
+	g.adjustStreamPriority(PriorityParam{
+		StreamDep: 11,
+		Weight:    212,
+	})
+
+	testRoots(t, rootTest{
+		roots:     sc.roots,
+		rootFirst: f,
+		rootLast:  b1,
+		l:         2,
+		n:         22,
+	})
+
+	testStreamNode(t, []streamTest{
+		streamTest{
+			st:        a,
+			depPrev:   m,
+			depNext:   d,
+			sibPrev:   nil,
+			sibNext:   k,
+			weight:    12,
+			weightSum: sumWeight(d),
+			n:         2,
+		},
+		streamTest{
+			st:        b,
+			depPrev:   nil,
+			depNext:   nil,
+			sibPrev:   i,
+			sibNext:   nil,
+			weight:    defaultWeight,
+			weightSum: 0,
+			n:         1,
+		},
+		streamTest{
+			st:        c,
+			depPrev:   g,
+			depNext:   j,
+			sibPrev:   nil,
+			sibNext:   nil,
+			weight:    20,
+			weightSum: sumWeight(j),
+			n:         14,
+		},
+		streamTest{
+			st:        d,
+			depPrev:   a,
+			depNext:   nil,
+			sibPrev:   nil,
+			sibNext:   nil,
+			weight:    defaultWeight,
+			weightSum: 0,
+			n:         1,
+		},
+		streamTest{
+			st:        e,
+			depPrev:   nil,
+			depNext:   nil,
+			sibPrev:   h,
+			sibNext:   nil,
+			weight:    255,
+			weightSum: 0,
+			n:         1,
+		},
+		streamTest{
+			st:        f,
+			depPrev:   nil,
+			depNext:   g,
+			sibPrev:   nil,
+			sibNext:   nil,
+			weight:    21,
+			weightSum: sumWeight(g),
+			n:         16,
+		},
+		streamTest{
+			st:        g,
+			depPrev
```

Comment:
maybe build the tree more explicitly with somthing like `StreamDep : a.id,` here? That might help when inserting/removing lines in the tests.